README.md

```diff
@@ -127,42 +126,48 @@ OakMap<K,V> oak = builder.build();
+
+For backward compatibility with applications that are already based on the use of ByteBuffers, Oak Buffers provide the transform method that atomically applies a transformation to the underlying ByteBuffer. For a more comprehensive code example please refer to the [usage](#usage) section. 
+
 
 ### Notes on data retrieval
-1. For better performance of data retrieval, OakMap supplies an OakBufferView of the OakMap. The OakBufferView provides the following four methods for data retrieval, whose result is presented as an OakRBuffer:
+1. For better performance of data retrieval, OakMap supplies a ZeroCopyMap interface of the OakMap:
+    `ZeroCopyMap<K, V> zc()` 
+    
+    The ZeroCopyMap interface provides the following four methods for data retrieval, whose result is presented as an OakRBuffer:
 	- `OakRBuffer get(K key)`
-	- `OakCloseableIterator<OakRBuffer> valuesIterator()`
-	- `OakCloseableIterator<Map.Entry<OakRBuffer, OakRBuffer>> entriesIterator()`
-	- `OakCloseableIterator<OakRBuffer> keysIterator()`
-2. Without the OakBufferView, OakMap's data can be directly retrieved via the following four methods:
-	- `V get(K key)`
-	- `OakCloseableIterator<V> valuesIterator()`
-	- `OakCloseableIterator<Map.Entry<K, V>> entriesIterator()`
-	- `OakCloseableIterator<K> keysIterator()`
+	- `Collection<OakRBuffer> values()`
+	- `Set<Map.Entry<OakRBuffer, OakRBuffer>> entrySet()`
+	- `Set<OakRBuffer> keySet()`
+2. Without ZeroCopyMap, OakMap's data can be directly retrieved via the following four methods:
+	- `V get(Object key)`
+	- `Collection<V> values()`
+	- `Set<Map.Entry<K, V>> entrySet()`
+	- `NavigableSet<K> keySet()`
 	
-	However, these direct methods return keys and/or values as Objects by applying deseriliazation (copy). This is costly,  and we strongly advice to use OakBufferView or OakTransformView to operate directly on the internal data representation.
-3. For further understanding of data retrieval via OakTransformView, please refer to the [Transformations](#transformations) section.
-4. Note that OakMap's iterators are `OakCloseableIterator` (extend AutoCloseable) so it is possible to reuse the memory previously referred by iterator. Be sure to use it within try-statement or call its close() method explicitly when iterator is no longer in use.
+	However, these direct methods return keys and/or values as Objects by applying deseriliazation (copy). This is costly,  and we strongly advice to use ZeroCopyMap to operate directly on the internal data representation.
+3. For examples of direct data manipulations, please refer to the [usage](#usage) section.
 
 ### Notes on data ingestion
-1. Data can be ingested and updated via the following five methods:
+1. Data can be ingested via the standard ConcurrentNavigableMap API.
```

Comment:
What about compute methods given by ConcurrentNavigableMap? Like:

default V computeIfPresent(K key,
                           BiFunction<? super K,? super V,? extends V> remappingFunction)

Do we support it anyhow?

Reply:
For non zero-copy operations we sometimes rely on default implementations in the map hierarchy.

Specifically, `computeIfPresent` has a default implementation in `ConcurrentMap` that uses `replace` and `remove`, which we do implement explicitly.